homework/index.js

```diff
@@ -1,46 +1,165 @@
 'use strict';
 
 {
-  function fetchJSON(url, cb) {
+  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
+  // defined for taking data from url
+  function fetchJSON(url, callBackFunction) {
     const xhr = new XMLHttpRequest();
     xhr.open('GET', url);
     xhr.responseType = 'json';
     xhr.onload = () => {
       if (xhr.status >= 200 && xhr.status <= 299) {
-        cb(null, xhr.response);
+        callBackFunction(null, xhr.response);
       } else {
-        cb(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
+        callBackFunction(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
       }
     };
-    xhr.onerror = () => cb(new Error('Network request failed'));
+    xhr.onerror = () => callBackFunction(new Error('Network request failed'));
     xhr.send();
   }
-
+  // defined for new element creat&append
   function createAndAppend(name, parent, options = {}) {
-    const elem = document.createElement(name);
-    parent.appendChild(elem);
+    const child = document.createElement(name);
+    parent.appendChild(child);
     Object.keys(options).forEach(key => {
       const value = options[key];
       if (key === 'text') {
-        elem.textContent = value;
+        child.textContent = value;
       } else {
-        elem.setAttribute(key, value);
+        child.setAttribute(key, value);
       }
     });
-    return elem;
+    return child;
+  }
+  // defined for error handling
+  function renderError(error) {
+    const root = document.getElementById('root');
+    createAndAppend('div', root, {
+      text: error.message,
+      class: 'alert',
+    });
+  }
+  // for creating select element
+  function createOptionElements(repositories, select) {
+    createAndAppend('option', select, {
+      text: 'Select a repository',
+      disabled: 'disabled',
+    });
+    repositories
+      .sort((a, b) => a.name.localeCompare(b.name))
+      .forEach((repository, index) => {
+        createAndAppend('option', select, {
+          text: repository.name,
+          value: index,
+        });
+      });
   }
 
-  function main(url) {
-    fetchJSON(url, (err, repositories) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+  function renderRepositories(repository) {
+    const leftBox = document.getElementById('left-box');
+    leftBox.innerHTML = '';
+    const rTable = createAndAppend('table', leftBox);
+    const tbody = createAndAppend('tbody', rTable, {
+      class: 'table_elms',
+    });
+    const tr1 = createAndAppend('tr', tbody);
+    createAndAppend('td', tr1, {
+      text: 'Repository      :',
+    });
+    createAndAppend('td', tr1, {
+      text: repository.name,
+    });
+    const tr2 = createAndAppend('tr', tbody);
+    createAndAppend('td', tr2, {
+      text: 'Description    :',
```

Comment:
These extra spaces between the label text and the colon are reduced to a single space when the page is rendered. There is no point of adding them. Even if this was not the case, the colons wouldn't line up anyway as the font used in the page has variable width characters, unlike the fixed font as used here in GitHub and in VSCode.